plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -307,12 +309,20 @@ private Schema toIcebergSchema(List<Column> columns)
         return new Schema(icebergSchema.asStructType().fields());
     }
 
-    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId)
+    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId, HiveStorageFormat storageFormat)
     {
         if (type instanceof ArrayType || type instanceof MapType || type instanceof RowType) {
             // TODO https://github.com/trinodb/trino/issues/17583 Add support for these complex types
             throw new TrinoException(NOT_SUPPORTED, "Migrating %s type is not supported".formatted(type));
         }
+
+        if (type.equals(TimestampType.TIMESTAMP_MICROS)) {
+            // TODO (https://github.com/trinodb/trino/pull/17221) Support for Avro
+            return switch (storageFormat) {
+                case ORC, PARQUET -> Types.TimestampType.withoutZone();
+                default -> throw new TrinoException(NOT_SUPPORTED, "Migrating timestamp(3) type with %s format is not supported. Use \"timestamp(6)\" instead.".formatted(storageFormat));
```

Comment:
`/defautt/AVRO` and move TODO:

```suggestion
            return switch (storageFormat) {
                case ORC, PARQUET -> Types.TimestampType.withoutZone();
                // TODO (https://github.com/trinodb/trino/pull/17221) Support for Avro
                AVRO -> throw new TrinoException(...);
```

Reply:
- the check isn't sufficient to block timesamp conversion in Avro tables -- the timestamp may be nested.
- why is message referring to `timestamp(3)` and `timestamp(6)`? what actually is the message asking the user to do?

Reply:
Nevertheless, just fyi, if we add explicit case for AVRO, we need to add default case anyway as it doesn't not fullfill all enums values from enum class HiveStorageFormat.